### ISSUE TITLE
details about training loss in pre_bc stage

### ISSUE THREAD

**Issue by Keirra0116** [NVlabs/catk#1]
Hi, I want to figure out some details about training loss in pre_bc stage. 

[catk/src/smart/metrics/cross_entropy.py, lines 83 to 91](https://github.com/NVlabs/catk/blob/main/src/smart/metrics/cross_entropy.py#L83-L91)
![image](https://github.com/user-attachments/assets/38fb7fb1-d059-4e73-bf45-a7dc918906f2)

I noticed that in the function `get_euclidean_targets`, the variable pred_pos is actually the discretized trajectory (the trajectory associated with a token). Specifically, when topk=1, the supervision signal (gt) is the token corresponding to the nearest tokenized trajectory to the ground truth (GT) trajectory.

From my understanding, this difference is fixed and does not dynamically change. If this is correct, then this approach seems to behave the same as the Smart when topk=1, as both use a fixed discretized ground truth token for supervision. Is that right? Thanks！  @zhejz

**Comment by Keirra0116** [comment]
Hi again, I have another question regarding the function `get_euclidean_targets` .

I’m still a bit confused about this part. It looks like gt_last_pos and gt_last_head represent the ground truth at the next time step after rolling the gt_pos and gt_head tensors. So, is the target calculated in this function the difference between the current prediction values (pred_pos, pred_head) and the next ground truth values (gt_last_pos, gt_last_head)?

Thanks for your help! @zhejz

**Comment by zhejz** [comment]
Yes, the pre-training via BC is the same as done in SMART.

> So, is the target calculated in this function the difference between the current prediction values (pred_pos, pred_head) and the next ground truth values (gt_last_pos, gt_last_head)?

Yes. This statement is correct. It's the "action". Specifically, it's a vector that points from the predicted state at t to the GT state at t+1.

For the pre-BC stage, the pred_pos/head and the GT_pos&head is identical. For closed-loop fine-tuning stage, the pred_pos&head will diverge.

**Comment by Keirra0116** [comment]
Thanks for your useful information!
